### PR TITLE
`TKEDissipationVerticalDiffusivity` (aka k-epsilon) turbulence closure

### LINKED ISSUE #2278: Implementing k-epsilon
Hi,
I just would like to give my humble contribution opening this issue.
I might not be able to do that because I am just a newbie in Julia, but it would be very nice to have k-epsilon as an option for turbulence closure.

Questions:
Where to start?
Any idea on how to implement that?

### PR DESCRIPTION
**Pull request by glwagner**
This PR implements a two-equation turbulent closure, which evolve turbulent kinetic energy and an additional "scale variable". Here we choose the scale variable dissipation; another name for this two-equation model is "k-epsilon". In Oceananigans it will be called the `TKEDissipationVerticalDiffusivity`.

This PR is a WIP; we still have to implement the stability functions and, perhaps more importantly, figure out how to make time-stepping stable. 

Closes #2278

### CODE REVIEW

File: src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/compute_tke_dissipation_diffusivity_fields.jl [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    return (; κu, κc, κe, κϵ, Le, Lϵ,
+            _tupled_tracer_diffusivity_fields,
+            _tupled_implicit_linear_coefficients)
+end        
+
+@inline viscosity(::FlavorOfKEpsilon, diffusivity_fields) = diffusivity_fields.κu
+@inline diffusivity(::FlavorOfKEpsilon, diffusivity_fields, ::Val{id}) where id =
+    diffusivity_fields._tupled_tracer_diffusivity_fields[id]
+
+function compute_diffusivities!(diffusivity_fields, closure::FlavorOfKEpsilon, model; parameters = :xyz)
```
Comment (francispoulin) [inline src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/compute_tke_dissipation_diffusivity_fields.jl:42]: You changed a lot of the names from `diffusivities` to `diffusivity_fields`, which looks better to me.

In this spirit, did you want to call this `compute_diffusivity_fields`?  I'm find either way, just a suggestion.

Comment (glwagner) [reply src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/compute_tke_dissipation_diffusivity_fields.jl:42]: yes eventually! that is a bigger change so I didn't do it yet

Comment (francispoulin) [reply src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/compute_tke_dissipation_diffusivity_fields.jl:42]: I am very excited to see this PR come to life.  Thanks for getting it started!

**Comment by navidcy** [comment]
btw this PR closes #2278

**Comment by glwagner** [comment]
nice

**Comment by glwagner** [comment]
Superceded by #3605

**Pull request closed without merging**